File: Beginner Study Guide/1342. Number of Steps to Reduce to Zero/code.java
class Solution {
    public int numberOfSteps(int num) {
        int numOfSteps = 0;

        while(num > 0) {
            boolean isEven = (num & 1) == 0;
            num = isEven ? num >>= 1 : num ^ 1;
            numOfSteps++;
        }

        return numOfSteps;
    }
}

File: Blind 75/347. Top K Frequent Elements/code.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        if n == 1: return nums

        num_freq = {}
        buckets = [[] for _ in range(n + 1)]
        output = []

        for num in nums:
            num_freq[num] = 1 + num_freq.get(num, 0)

        for num, freq in num_freq.items():
            buckets[freq].append(num)

        for i in range(n, 0, -1):
            for val in buckets[i]:
                output.append(val)
                if len(output) == k: return output
